src/main/java/com/powsybl/openloadflow/sa/DcSecurityAnalysis.java

```diff
@@ -73,4 +75,26 @@ protected DcLoadFlowEngine createLoadFlowEngine(DcLoadFlowContext context) {
     protected PostContingencyComputationStatus postContingencyStatusFromLoadFlowResult(DcLoadFlowResult result) {
         return result.isSuccess() ? PostContingencyComputationStatus.CONVERGED : PostContingencyComputationStatus.FAILED;
     }
+
+    @Override
+    protected Consumer<DcLoadFlowParameters> createParametersResetter(DcLoadFlowParameters parameters) {
+        boolean oldDistributedSlack = parameters.isDistributedSlack();
+        LoadFlowParameters.BalanceType oldBalanceType = parameters.getBalanceType();
+        List<DcOuterLoop> oldOuterLoops = List.copyOf(parameters.getOuterLoops());
+        return p -> {
+            p.setDistributedSlack(oldDistributedSlack);
+            p.setBalanceType(oldBalanceType);
+            p.setOuterLoops(oldOuterLoops);
+        };
+    }
+
+    @Override
+    protected void applyContingencyParameters(DcLoadFlowParameters parameters, ContingencyLoadFlowParameters contingencyParameters, LoadFlowParameters loadFlowParameters, OpenLoadFlowParameters openLoadFlowParameters) {
+        DcOuterLoopConfig outerLoopConfig = DcOuterLoopConfig.findOuterLoopConfig()
```

Comment:
same here, probably caching needed for `DcOuterLoopConfig.findOuterLoopConfig()`